src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+      >
+        {this.translations.text.prev}
+      </button>
+    );
+  };
+
+  #renderNextControl = () => {
+    return (
+      <button
+        part={tokenMap({
+          [PAGINATOR_PARTS_DICTIONARY.NEXT]: true,
+          [PAGINATOR_PARTS_DICTIONARY.DISABLED]:
+            this.actualPage === this.totalPages
+        })}
+        disabled={this.actualPage === this.totalPages}
+        onClick={this.#handleNext}
+      >
+        {this.translations.text.next}
+      </button>
+    );
+  };
+
+  #renderNavigationControls = () => {
+    return (
+      <ol
+        class="pages"
+        part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.PAGES]: true })}
+        onClick={this.#handleSelectedPageChange}
+      >
+        {this.#getPaginationRange().map((page, index) => {
+          if (page > 0) {
+            // Validate if pagination has hyperlink
+            if (this.hasHyperlink) {
+              return (
+                <li>
+                  <a
+                    key={index}
+                    part={tokenMap({
+                      [PAGINATOR_PARTS_DICTIONARY.PAGE_ACTIVE]:
+                        this.actualPage === page
+                    })}
+                    href={this.model[page - 1]}
+                  >
+                    {page}
+                  </a>
+                </li>
+              );
+            }
+            // Validate if pagination has urlMapping, if true, return anchor element
+            if ((this.model as PaginatorRenderNumericModel)?.urlMapping) {
+              return (
+                <li>
+                  <a
+                    key={index}
+                    id={page.toString()}
+                    part={tokenMap({
+                      [PAGINATOR_PARTS_DICTIONARY.PAGE_ACTIVE]:
+                        this.actualPage === page
+                    })}
+                    href={(
+                      this.model as PaginatorRenderNumericModel
+                    ).urlMapping(page)}
+                  >
+                    {page}
+                  </a>
+                </li>
+              );
+            }
+            // Else, return a button with page number as value
+            return (
+              <li>
+                <button
+                  key={index}
+                  part={tokenMap({
+                    [PAGINATOR_PARTS_DICTIONARY.PAGE_ACTIVE]:
+                      this.actualPage === page
+                  })}
+                  value={page}
+                >
+                  {page}
+                </button>
+              </li>
+            );
+          }
+          // if page is "-1", it renders a span with the ellipsis
+          return (
+            <li>
+              <span
+                key={index}
+                part={tokenMap({
+                  [PAGINATOR_PARTS_DICTIONARY.ELLIPSIS]: true
+                })}
+              >
+                {this.translations.text.ellipsis}
+              </span>
+            </li>
+          );
+        })}
```

Comment:
It would be nice to move this render logic into a separated function.

```suggestion
        {this.#getPaginationRange().map(this.#renderPaginatorItem)}
```